tests/cpp/test_multidevice_overlap.cpp

```diff
@@ -472,4 +472,134 @@ TEST_F(
   }
 }
 
+class AGOverlapTest : public MultiDeviceTest {
+ protected:
+  OverlapTestParams params;
+
+  int64_t num_devices_;
+  int64_t my_device_index_;
+  std::vector<int64_t> all_devices_;
+  at::Tensor ta_, tb_, tc_;
+  at::Tensor ta_sharded_, tc_sharded_;
```

Comment:
nit: Since A is sharded, and B and C are unsharded, I would use the notations `ta_unsharded_` and `tc_sharded_` in contrast with `ta_, tb_, tc_`